project/src/store/middlewares/redirect.ts

```diff
@@ -0,0 +1,16 @@
+import browserHistory from '../../browser-history';
+import {Middleware} from 'redux';
+import {reducer} from '../reducer';
+
+type Reducer = ReturnType<typeof reducer>;
+
+export const redirect: Middleware<unknown, Reducer> =
+  (_store) =>
+    (next) =>
+      (action) => {
+        if (action.type === 'app/redirectToRoute') {
```

Comment:
```suggestion
        if (action.type === action.type) {
```
чтобы не дублировать эту строку, лучше так

Reply:
а так разве не всегда будет `true` или тут enum для экшнов нужно создать?

Reply:
у каждого экшена, что ты создаешь через createAction, есть это поле

Reply:
а, я немного не так написасл. нужно взять именно экшен с файла с экшенами, а не тот который нам приходит в аргумент мидовары